conversation_event.go

```diff
@@ -0,0 +1,37 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+type EventParams struct {
+	// Type identifies the type of event (start-typing, stop-typing, joined, left)
+	Type string `json:"name"`
```

Comment:
What do you think of adding an enum for these? Mostly to serve as inline documentation.

```suggestion
// ConversationEventType describes an event that occurred within the conversation.
type ConversationEventType string

const (
	// ConversationEventTypeStartTyping means the customer or human agent started
	// typing.
	ConversationEventTypeStartTyping ConversationEventType = "start-typing"

	// ConversationEventTypeStopTyping means the customer or human agent stopped
	// typing.
	ConversationEventTypeStopTyping ConversationEventType = "stop-typing"

	// ConversationEventTypeJoin means the customer or human agent joined the
	// conversation.
	ConversationEventTypeJoin ConversationEventType = "join"

	// ConversationEventTypeLeave means the customer or human agent left the
	// conversation.
	ConversationEventTypeLeave ConversationEventType = "leave"
)

type EventParams struct {
	// Type identifies the type of event (start-typing, stop-typing, join, leave).
	Type ConversationEventType `json:"name"`
```